qiskit/synthesis/multi_controlled/mcmt_vchain.py

```diff
@@ -0,0 +1,49 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Synthesis for multiple-control, multiple-target Gate."""
+
+from __future__ import annotations
+
+from qiskit.circuit import QuantumCircuit, Gate
+from qiskit._accelerate.synthesis.multi_controlled import mcmt_v_chain
+
+
+def synth_mcmt_vchain(
+    gate: Gate, num_ctrl_qubits: int, num_target_qubits: int, ctrl_state: int | None = None
+) -> QuantumCircuit:
+    """Synthesize MCMT using a V-chain.
+
+    This uses a chain of CCX gates, using ``num_ctrl_qubits - 1`` auxiliary qubits.
+
+    For example, a 3-control and 2-target H gate will be synthesized as::
+
+        q_0: ──■────────────────────────■──
+               │                        │
+        q_1: ──■────────────────────────■──
+               │                        │
+        q_2: ──┼────■──────────────■────┼──
+               │    │  ┌───┐       │    │
+        q_3: ──┼────┼──┤ H ├───────┼────┼──
+               │    │  └─┬─┘┌───┐  │    │
+        q_4: ──┼────┼────┼──┤ H ├──┼────┼──
+             ┌─┴─┐  │    │  └─┬─┘  │  ┌─┴─┐
+        q_5: ┤ X ├──■────┼────┼────■──┤ X ├
+             └───┘┌─┴─┐  │    │  ┌─┴─┐└───┘
+        q_6: ─────┤ X ├──■────■──┤ X ├─────
+                  └───┘          └───┘
+
```

Comment:
A side question (not fully relevant for this PR): the definition of the MCMT gate requires that both target qubits correspond to the same single-qubit gate (e.g. H in the example above). Would the construct remain correct if the single-qubit gates were different (e.g. H on one target qubit and S on another target qubit)? 

In fact, if we were to compose two copies of the construction above 
```(CCX-chain -- U -- inverse-CCX-chain) -- (CCX-chain -- V -- inverse-CCX-chain)```,
then we would get
```(CCX-chain -- U  -- V -- inverse-CCX-chain)```, so the only benefit of defining an MCMT-gate vs. several regular multi-controlled gates is that the synthesis would necessarily create the same chain of CCX gates and allow cancellation. Am I correct about this?

Reply:
Yes 🙂 As discussed offline, we might want to make the V-chain available as generic control algorithm, akin to the current `add_control`. This would then allow HLS to synthesize any controlled gates or annotated controlled gates with either of the two decompositions.